common/python/utility/file_utils.py

```diff
@@ -70,6 +70,62 @@ def read_json_file(input_file, data_dir=['./', '../', '/']):
     return "".join(input_lines)
 
 
+# -----------------------------------------------------------------
+def read_file(input_file):
+    """
+    Utility function to read a file
+
+    Parameters:
+        @param input_file - Path of file to be read
+    Returns:
+        @returns data - Contents of the file if valid.
+                        None, otherwise.
+
+    """
+    data = ""
+    if len(input_file) != 0 and os.path.exists(input_file):
+        logging.debug("Reading the file %s", input_file)
+        try:
+            with open(input_file, 'r') as data_file:
+                data = data_file.read()
+        except IOError as e:
```

Comment:
Is IOError the only exception that can happen ? We can also catch a generic Exception to be on safer side.

Reply:
Makes sense. Will see if read/write can throw some other error. open() is expected to raise IOError.